File: CS/DatabaseUserSettings/BusinessObjects/XPUserSettingsAspect.cs
using DevExpress.ExpressApp.Model;
using System;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.Validation;

namespace DatabaseUserSettings.BusinessObjects {
    [ImageName("ModelEditor_ModelMerge")]
    [ModelDefault("Caption", "User Settings Aspect")]
    public class XPUserSettingsAspect : XPObject, IDatabaseUserSettingsAspect {
        public XPUserSettingsAspect(Session session) : base(session) { }
        [Association]
        [RuleRequiredField(null, DefaultContexts.Save)]
        public XPUserSettings Owner {
            get { return GetPropertyValue<XPUserSettings>("Owner"); }
            set { SetPropertyValue<XPUserSettings>("Owner", value); }
        }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
        [Size(SizeAttribute.Unlimited)]
        public string Xml {
            get { return GetPropertyValue<string>("Xml"); }
            set { SetPropertyValue<string>("Xml", value); }
        }
        IDatabaseUserSettings IDatabaseUserSettingsAspect.Owner {
            get { return Owner; }
            set { Owner = value as XPUserSettings; }
        }
    }
}

File: VB/DatabaseUserSettings/BusinessObjects/XPUserSettingsAspect.vb
Imports DevExpress.ExpressApp.Model
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.Validation

Namespace DatabaseUserSettings.BusinessObjects
    <ImageName("ModelEditor_ModelMerge"), ModelDefault("Caption", "User Settings Aspect")> _
    Public Class XPUserSettingsAspect
        Inherits XPObject
        Implements IDatabaseUserSettingsAspect

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        <Association, RuleRequiredField(Nothing, DefaultContexts.Save)> _
        Public Property Owner() As XPUserSettings
            Get
                Return GetPropertyValue(Of XPUserSettings)("Owner")
            End Get
            Set(ByVal value As XPUserSettings)
                SetPropertyValue(Of XPUserSettings)("Owner", value)
            End Set
        End Property
        Public Property Name() As String Implements IDatabaseUserSettingsAspect.Name
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Name", value)
            End Set
        End Property
        <Size(SizeAttribute.Unlimited)> _
        Public Property Xml() As String Implements IDatabaseUserSettingsAspect.Xml
            Get
                Return GetPropertyValue(Of String)("Xml")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Xml", value)
            End Set
        End Property
        Private Property IDatabaseUserSettingsAspect_Owner() As IDatabaseUserSettings Implements IDatabaseUserSettingsAspect.Owner
            Get
                Return Owner
            End Get
            Set(ByVal value As IDatabaseUserSettings)
                Owner = TryCast(value, XPUserSettings)
            End Set
        End Property
    End Class
End Namespace
